src/main/java/org/craftercms/studio/impl/v2/service/security/GroupServiceImpl.java

```diff
@@ -109,8 +125,36 @@ public Group getGroup(long groupId) throws ServiceLayerException {
     @Override
     @HasPermission(type = DefaultPermission.class, action = "update_groups")
     public void removeGroupMembers(long groupId, List<Long> userIds, List<String> usernames)
-        throws ServiceLayerException, UserNotFoundException {
-        securityProvider.removeGroupMembers(groupId, userIds, usernames);
+            throws ServiceLayerException, UserNotFoundException {
+        Group g = getGroup(groupId);
+        generalLockService.lock(REMOVE_SYSTEM_ADMIN_MEMBER_LOCK);
+        try {
+            if (g.getName().equals(SYSTEM_ADMIN_GROUP)) {
+                List<User> members = getGroupMembers(groupId, 0, Integer.MAX_VALUE, StringUtils.EMPTY);
+                if (CollectionUtils.isNotEmpty(members)) {
+                    List<User> membersAfterRemove = new ArrayList<User>();
+                    membersAfterRemove.addAll(members);
+                    members.forEach(m -> {
+                        if (CollectionUtils.isNotEmpty(userIds)) {
+                            if (userIds.contains(m.getId())) {
+                                membersAfterRemove.remove(m);
+                            }
+                        }
+                        if (CollectionUtils.isNotEmpty(usernames)) {
+                            if (usernames.contains(m.getUsername())) {
+                                membersAfterRemove.remove(m);
+                            }
+                        }
+                    });
+                    if (CollectionUtils.isEmpty(membersAfterRemove)) {
+                        throw new ServiceLayerException("Removing all users from System Admin group is not allowed");
```

Comment:
Change the wording to: "Removing all members of the System Admin group is not allowed. We must have at least one system administrator."